File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.c
int balancedStringSplit(char * s){
    int n = 0;
    int i = 0;
    int count = 0;
    
    while(s[i]){
        if(s[i] == 'R')
            n++;
        else
            n--;
        if(n == 0)
            count++;
        i++;
    }
    return count;
}

File: 1221-split-a-string-in-balanced-strings/1221-split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int c = 0;
        int n = 0;
        
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == 'R')
                n++;
            else
                n--;
            if(n == 0)
                c++;
        }
        return c;
    }
}